File: kora-app-annotation-processor/src/test/java/ru/tinkoff/kora/kora/app/annotation/processor/app/AppWithAllOfValueOf.java
package ru.tinkoff.kora.kora.app.annotation.processor.app;

import ru.tinkoff.kora.application.graph.All;
import ru.tinkoff.kora.application.graph.ValueOf;
import ru.tinkoff.kora.common.KoraApp;
import ru.tinkoff.kora.common.annotation.Root;

import java.util.Objects;

@KoraApp
public interface AppWithAllOfValueOf {
    @Root
    default Class1 class1(All<ValueOf<Class2>> cls) {
        for (var cl : cls) {
            Objects.requireNonNull(cl.get());
        }
        return new Class1();
    }

    default Class2 class2() {
        return new Class2();
    }

    class Class1 {}

    class Class2 {}

}

File: kora-app-symbol-processor/src/test/kotlin/ru/tinkoff/kora/kora/app/ksp/app/AppWithAllOfValueOf.kt
package ru.tinkoff.kora.kora.app.ksp.app

import ru.tinkoff.kora.application.graph.All
import ru.tinkoff.kora.application.graph.ValueOf
import ru.tinkoff.kora.common.KoraApp
import ru.tinkoff.kora.common.annotation.Root
import java.util.*

@KoraApp
interface AppWithAllOfValueOf {
    @Root
    fun class1(cls: All<ValueOf<Class2>>): Class1 {
        for (cl in cls) {
            Objects.requireNonNull(cl.get())
        }
        return Class1()
    }

    fun class2(): Class2 {
        return Class2()
    }

    class Class1
    class Class2
}
